github/git_commits.go

```diff
@@ -87,6 +90,29 @@ func (s *GitService) GetCommit(ctx context.Context, owner string, repo string, s
 	return c, resp, nil
 }
 
+// ListCommits fetches latest commits
```

Comment:
Please make this a complete sentence (ending with a period) for the sake of the auto-generated godocs.

Please also comment on the behavior of the `sha` parameter.  What happens if it is `""`?
Does it list commits _after_ the SHA?  Does it include the SHA in the list?
Inquiring minds want to know!  :smile:

Also, please send a request to GitHub developer tech support at support@github.com and request them to document this endpoint.  In the meantime, we can proceed with this PR, but when they respond, we will want to add a link to the official GitHub API docs like the other endpoints in this repo. It could either be in this PR or in a follow-on PR.  Usually they are incredibly fast in responding to requests like this.  Thank you!